### PR TITLE
m_Do_audio

### PR DESCRIPTION
**Pull request by jdflyer**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,7 @@ ASFLAGS := -mgekko -I include
 LDFLAGS := -map $(MAP) -fp hard -nodefaults -w off
 
 # Compiler flags
-CFLAGS  += -Cpp_exceptions off -proc gekko -fp hard -O3 -nodefaults -msgstyle gcc -enum int $(INCLUDES)
+CFLAGS  += -Cpp_exceptions off -proc gekko -fp hard -O3 -nodefaults -msgstyle gcc -maxerrors 5 -enum int $(INCLUDES)
```
Comment (barzamin) [inline Makefile:85]: :)

File: include/Z2AudioLib/Z2AudioMgr/Z2AudioMgr.h [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
+    Z2AudioMgr* setOutputMode(unsigned long outputMode); 
+    u32 resetProcess(u32 param_1, bool param_2);
+    bool hasReset(void) const;
+    void resetRecover(void);
+    void init(JKRSolidHeap*, unsigned long, void*, JKRArchive*);
 
-extern "C" {
-void Z2AudioMgr_NS_setOutputMode(void*, unsigned long);
-u32 Z2AudioMgr_NS_hasReset(void*);
+    u8 p1[0x4a4];
```
Comment (barzamin) [inline include/Z2AudioLib/Z2AudioMgr/Z2AudioMgr.h:16]: consider using a different prefix as `p` is typically used for function arguments by nintendo

File: include/m_Do/m_Do_audio/m_Do_audio.h [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#ifndef __M_DO_AUDIO_H_
+#define __M_DO_AUDIO_H_
+#include "global.h"
+#include "Z2AudioLib/Z2AudioMgr/Z2AudioMgr.h"
+
+class mDoAud_zelAudio_c{ //Based on Z2AudioMgr
```
Comment (barzamin) [inline include/m_Do/m_Do_audio/m_Do_audio.h]: by "based on": is it actually subclassed in the game?

File: include/m_Do/m_Do_audio/m_Do_audio.h [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+#include "Z2AudioLib/Z2AudioMgr/Z2AudioMgr.h"
+#include "global.h"
+
+class mDoAud_zelAudio_c {  // Based on Z2AudioMgr
+public:
+    void reset(void);
+    ~mDoAud_zelAudio_c();
+};
+
+class JAISeCategoryMgr {
```
Comment (barzamin) [inline include/m_Do/m_Do_audio/m_Do_audio.h:12]: should be decl'd in `JAISeMgr.cpp`

File: include/m_Do/m_Do_audio/m_Do_audio.h (hunk 3 above)
Comment (jdflyer) [reply include/m_Do/m_Do_audio/m_Do_audio.h]: I didn't do enough research to find out if it was, so I put that comment in there for future reference.

**Review by notyourav — APPROVED** [review]

**Pull request merged**